### PR TITLE
ssl-nginx: Add new LMS variant to serve SSL through nginx

### PR DESCRIPTION
**Pull request by antoviaque**
Allow to use nginx to serve the LMS through SSL, when an external proxy (like the elastic load balancer) doesn't already handle it.

Deactivated by default, add `lms-ssl` to `nginx_sites` to activate it.

### CODE REVIEW

File: playbooks/roles/nginx/tasks/ssl.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: nginx | Install SSL Certificate
+  copy: src="{{secure_dir}}/files/ssl/lms.crt" dest="/etc/ssl/certs/lms.crt" owner=root group=root mode=0640
+
+- name: nginx | Install SSL Certificate - Key
+  copy: src="{{secure_dir}}/files/ssl/lms.key" dest="/etc/ssl/private/lms.key" owner=root group=root mode=0640
+
```
Comment (carsongee) [inline playbooks/roles/nginx/tasks/ssl.yml:7]: Can you add a CA cert here as well?

**Comment by carsongee** [comment]
This is a good start.  We are using SSL at MITx (both client and server), and it would be nice to get that merged in.  I still think we need to add the nginx template overrides in though, since for us there is a whole lot more to configure to get ssl going that I'm sure is more specialized than is needed for everyone else.

File: playbooks/roles/nginx/tasks/ssl.yml (hunk 1 above)
Comment (antoviaque) [inline playbooks/roles/nginx/tasks/ssl.yml:7]: I currently include the CA cert(s) directly in the `.crt` file (appended, one after the other) - this wouldn't work in your case?

**Comment by antoviaque** [comment]
If you want needed I can add extra options - as long as they can have sensible defaults it's good with me : )

**Comment by carsongee** [comment]
It turns out I meant ssl_client_certificate and not CA, which 99.9% of people won't use and thus won't have.  So we'd have to add some conditional to check for that.  Because of that I'm still thinking we'll have to do a full override of the template as proposed with #339 to handle our specialized needs. It may be also be more appropriate to do this that way rather than adding an extra variant by having a template that has both the ssl and plain http in it for your lms nginx template, but having a split http/server side only https lms is going to be a way more common use case than what we're doing.  I'm happy to go either way honestly though, so I'll either leave this up to another reviewer, or we can discuss at the next hangout ;) We could always go both ways too.

**Comment by sasasaftic** [comment]
Can we activate ssl on edx after edx setup... using  /edx/etc/server-vars.yml?
(as we can for runnig our edx fork - https://github.com/edx/configuration/wiki/edX-Managing-the-Production-Stack#using-an-edx-platform-fork)
Or do I have to run whole ansible again?

**Comment by hkim823** [comment]
@antoviaque and @carsongee is this still a valid PR at this point?  Let us know

**Comment by carsongee** [comment]
This should be good to close as https://github.com/edx/configuration/pull/590 added SSL support.

**Pull request closed without merging**